lib/util/createHash.js

```diff
@@ -33,13 +34,19 @@ class BulkUpdateDecorator {
 		if(this.buffer.length > 0) {
 			this.hash.update(this.buffer);
 		}
+		if(this.algorithm === "xxhashjs") {
+			// See: https://github.com/pierrec/js-xxhash/blob/master/examples/from_file64.js
+			return this.hash.digest().toString(16);
+		}
 		return this.hash.digest(encoding);
 	}
 }
 
 module.exports = algorithm => {
 	switch(algorithm) {
-		// TODO add non-cryptographic algorithm here
-		default: return new BulkUpdateDecorator(require("crypto").createHash(algorithm));
+		case "xxhashjs":
+			return new BulkUpdateDecorator(require("xxhashjs").h64().init(), algorithm);
```

Comment:
Why `h64` instead of `h32`? The later is more efficient.

Reply:
hmm for some reason i thought it was opposite, updating. Thank you @ooflorent